# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/JSObject.h
# Commit: e15f1a6f5a38
# Full Hash: e15f1a6f5a38b276216c77f2827945b6f5ec4ea9
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 9 - Move proto from ObjectGroup to BaseShape. r=jonco
#   
#   The ObjectGroup still stores the proto as well but we don't really use it anymore.
#   This is simpler than removing the group's proto now because that would result in
#   "empty" groups and the group table then doesn't make sense anymore.
# ==============================================================================

diff -r 95fa0ec36f7a -r e15f1a6f5a38 js/src/vm/JSObject.h
--- a/js/src/vm/JSObject.h	Fri Mar 05 19:10:08 2021 +0000
+++ b/js/src/vm/JSObject.h	Fri Mar 05 19:10:08 2021 +0000
@@ -166,6 +166,13 @@
     return shape()->hasObjectFlag(flag);
   }
 
+  // Change this object's shape for a prototype mutation.
+  //
+  // Note: this does not reshape the proto chain to invalidate shape
+  // teleporting, check for an immutable proto, etc.
+  static bool setProtoUnchecked(JSContext* cx, JS::HandleObject obj,
+                                js::Handle<js::TaggedProto> proto);
+
   // An object is a delegate if it is (or was) another object's prototype.
   // Optimization heuristics will make use of this flag.
   // See: ReshapeForProtoMutation, ReshapeForShadowedProp
@@ -299,7 +306,7 @@
    *    the proto.
    */
 
-  js::TaggedProto taggedProto() const { return group()->proto(); }
+  js::TaggedProto taggedProto() const { return shape()->proto(); }
 
   bool uninlinedIsProxyObject() const;
 